internal/controller/dashboard_controller.go

```diff
@@ -206,23 +206,13 @@ func (r DashboardReconciler) configureDashboard(ctx context.Context, dashboardCM
 		return nil
 	}
 
-	// We always switch back to the shared org
-	defer func() {
-		if _, err = r.GrafanaAPI.SignedInUser.UserSetUsingOrg(grafana.SharedOrg.ID); err != nil {
-			logger.Error(err, "failed to change current org for signed in user")
-		}
-	}()
-
 	// Switch context to the dashboards-defined org
 	organization, err := grafana.FindOrgByName(r.GrafanaAPI, dashboardOrg)
 	if err != nil {
 		logger.Error(err, "failed to find organization", "organization", dashboardOrg)
 		return errors.WithStack(err)
 	}
-	if _, err = r.GrafanaAPI.SignedInUser.UserSetUsingOrg(organization.ID); err != nil {
-		logger.Error(err, "failed to change current org for signed in user")
-		return errors.WithStack(err)
-	}
+	grafanaAPIWithOrgID := r.GrafanaAPI.WithOrgID(organization.ID)
```

Comment:
This function does not work, that is why we moved to this mechanism. Also, there is not race condition to fix here

Reply:
Merged in https://github.com/giantswarm/observability-operator/pull/252